components/sup/src/cli_test_helpers.rs

```diff
@@ -0,0 +1,153 @@
+use std::fmt;
+use std::iter::FromIterator;
+use clap::{App, ArgMatches};
+
+#[derive(Debug)]
+pub enum Expectation<'a> {
+    Value(&'a str),
+    Values(Vec<String>)
+}
+
+use self::Expectation::{Value, Values};
+
+#[derive(Debug)]
+enum CliTestError {
+    ValueMismatch(String, String, String),
+    MultiValueMismatch(String, Vec<String>, Vec<String>),
+    MissingFlag(String),
+    MissingSubcmd(),
+    ClapError(String)
+}
+
+impl fmt::Display for CliTestError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let content = match self {
+            CliTestError::ValueMismatch(flag, expected, actual) => format!(
+                r#"
+'{}' validation failed:
+  Expected: '{}'
+  Actual:   '{}'
+"#,
```

Comment:
This is fine, but just FYI, there's another option you can use to keep the indentation in the code nicer. Since a backslash can be used to escape leading whitespace:
```rust
        println!("'{}' validation failed:\n  \
                  Expected: '{}'\n  \
                  Actual:   '{}'",
                  "foo", "bar", "baz");
```
yields:
```
'foo' validation failed:
  Expected: 'bar'
  Actual:   'baz'
```